site/documentation/content/user-guide/coap-adapter.md

```diff
@@ -0,0 +1,231 @@
+
+## Message Limits
+
+Before accepting any telemetry or event or command messages, the CoAP adapter verifies that the configured [message limit] ({{< relref "/concepts/resource-limits.md" >}}) is not exceeded. If the limit is exceeded then the incoming message is discarded with the status code `429 Too Many Requests`. 
+
+## Publish Telemetry Data (authenticated Device)
+
+The device is authenticated using PSK.
+
+* URI: `/telemetry`
+* Method: `POST`
+* Type:
+  * `CON`: *at least once* (`1`) QoS levels
+  * `NON`: *at most once* (`0`) QoS levels
+* Request Options:
+  * (required) `content-type`: The type of payload contained in the request body.
+  * (optional) `URI-query: hono-ttd`: The number of seconds the device will wait for the response.
+  * (optional) `URI-query: empty`: If the request doesn't contain payload.
+* Request Body:
+  * (required) Arbitrary payload encoded according to the given content type.
+* Response Options:
+  * (optional) `content-type`: A media type describing the semantics and format of payload contained in the response body. This header will only be present if the response contains a command to be executed by the device which requires input data.
+  * (optional) `location-query: hono-command`: The name of the command to execute. This header will only be present if the response contains a command to be executed by the device.
+  * (optional) `location-path: ('command'|'command_response/'<command-request-id>)`: The location path contains `'command'` for one-way-commands, or `'command_response' / <command-request-id>`  for commands expecting  a response. In case of a expected response, the location-path contains exactly the URI-path expected for the response.
+* Response Body:
+  * (optional) Arbitrary data serving as input to a command to be executed by the device, if status code is 200.
+  * (optional) Error details, if status code is >= 400.
+* Status Codes:
+  * 205 (CONTENT): The telemetry data has been accepted for processing. The response contains a command for the device to execute.
+  * 204 (CHANGED): The telemetry data has been accepted for processing. Note that if the message type is `NON` (*at most once* semantics),
+    this status code does **not** mean that the message has been delivered to any potential consumer. However, if the message type is `CON` (*at least once* semantics), then the adapter waits for the message to be delivered and accepted by a downstream consumer before responding with
+    this status code.
+  * 400 (Bad Request): The request cannot be processed. Possible reasons for this include:
+        * The content type header is missing.
+        * The request body is empty.
+        * The QoS header value is invalid.
+  * 401 (Unauthorized): The request cannot be processed because the request does not contain valid credentials.
```

Comment:
the request itself never contains credentials, does it?

Reply:
If the DTLS handshakes fails, no response is sent at all. So the 401 will only be sent back, if some hono-internal component reports that. I can't see, how that source could be verified efficiently, if it reports that error or not and what will be that cause for 401.